tests/test_closing.py

```diff
@@ -3631,3 +3631,71 @@ def test_close_weight_estimate(node_factory, bitcoind):
     signed_weight = int(bitcoind.rpc.decoderawtransaction(tx)['weight'])
     assert signed_weight <= actual_weight
     assert signed_weight >= actual_weight - 2
+
+
+@pytest.mark.developer("needs dev_disconnect")
+def test_onchain_close_upstream(node_factory, bitcoind):
+    """We send an HTLC, and peer unilaterally closes: do we close
+upstream?  Aka
+https://github.com/ElementsProject/lightning/issues/4649
+
+    """
+    l1, l2, l3 = node_factory.line_graph(3, wait_for_announce=True,
+                                         opts=[{'feerates': (7500, 7500, 7500, 7500)},
+                                               # Second commitment_signed is to l3
+                                               {'disconnect': ['xWIRE_COMMITMENT_SIGNED*2'],
+                                                # We want htlc killed by timeout, not a close due to ping timer.
+                                                'dev-no-ping-timer': None},
+                                               {'dev-no-ping-timer': None}])
+
+    ph1 = l3.rpc.invoice(msatoshi="10000sat", label='x1', description='desc2')['payment_hash']
+    ph2 = l3.rpc.invoice(msatoshi="10000sat", label='x2', description='desc2')['payment_hash']
+
+    route = l1.rpc.getroute(l3.info['id'], 1, 1)['route']
+
+    # Start a payment
+    l1.rpc.sendpay(route, ph1)
+
+    # l3 sends commitment_signed, then silence.
+    l2.daemon.wait_for_log('dev_disconnect: xWIRE_COMMITMENT_SIGNED')
+
+    # Send another payment, this times out.
+    l1.rpc.sendpay(route, ph2)
+
+    # This can take 30 seconds...
+    l2.daemon.wait_for_log('Adding HTLC 1 too slow: killing connection',
+                           timeout=TIMEOUT + 30)
+    l2.daemon.wait_for_log('Failing HTLC 1 due to peer death')
+
+    with pytest.raises(RpcError, match=r'WIRE_TEMPORARY_CHANNEL_FAILURE \(reply from remote\)'):
+        l1.rpc.waitsendpay(ph2, timeout=TIMEOUT)
+    
```

Comment:
There should be extra whitespace that I don't see here.

A simple space to stop the CI :/